### PR TITLE
Corrupted state DataObject::Peak

### PR DESCRIPTION
**Pull request by OwenArnold**
See [#11056](http://trac.mantidproject.org/mantid/ticket/11056) for issue.

### CODE REVIEW

**Comment by martyngigg** [comment]
There seem to be quite a few unit test failures

**Comment by OwenArnold** [comment]
Jenkins retest this please

**Comment by OwenArnold** [comment]
Jenkins retest this please.

**Comment by OwenArnold** [comment]
Jenkins retest this please

**Comment by OwenArnold** [comment]
Jenkins retest this please. Internal compiler error for icpc that I can't reproduce on last pull-request build.

File: Code/Mantid/Framework/Crystal/test/AddPeakHKLTest.h [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+      Mantid::Geometry::Goniometer goniometer; // identity
+      V3D hkl = qLab / (2 * M_PI); // Given our settings above, this is the simplified relationship between qLab and hkl.
+
+      // Now create a peaks workspace around the simple fictional instrument
+      PeaksWorkspace_sptr ws = boost::make_shared<PeaksWorkspace>();
+      ws->setInstrument(minimalInstrument);
+      ws->mutableSample().setOrientedLattice(&orientedLattice);
+      ws->mutableRun().setGoniometer(goniometer, false);
+
+      Mantid::API::AnalysisDataService::Instance().add("peaks_ws",ws);
```
Comment (martyngigg) [inline Code/Mantid/Framework/Crystal/test/AddPeakHKLTest.h]: Is there a real need to add the workspace to the ADS here? If the algorithm is set as a child then workspace can be retrieved from the algorithm, avoiding touching the global store.

**Comment by martyngigg** [comment]
The code changes look good now and the system tests look fine.

**Pull request merged**